azurerm/internal/services/loganalytics/log_analytics_datasource_linux_performance_collection_resource.go

```diff
@@ -0,0 +1,187 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/loganalytics/parse"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmLogAnalyticsDataSourceLinuxPerformanceCollection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLogAnalyticsDataSourceLinuxPerformanceCollectionCreateUpdate,
+		Read:   resourceArmLogAnalyticsDataSourceLinuxPerformanceCollectionRead,
+		Update: resourceArmLogAnalyticsDataSourceLinuxPerformanceCollectionCreateUpdate,
+		Delete: resourceArmLogAnalyticsDataSourceLinuxPerformanceCollectionDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImportThen(func(id string) error {
+			_, err := parse.LogAnalyticsDataSourceID(id)
+			return err
+		}, importLogAnalyticsDataSource(operationalinsights.LinuxPerformanceCollection)),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"workspace_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     ValidateAzureRmLogAnalyticsWorkspaceName,
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+type dataSourceLinuxPerformanceCollectionProperty struct {
+	State string `json:"state"`
+}
+
+func resourceArmLogAnalyticsDataSourceLinuxPerformanceCollectionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.DataSourcesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	workspaceName := d.Get("workspace_name").(string)
+	perfCollectionState := "Disabled"
+
+	if v, ok := d.GetOk("enabled"); ok {
+		if enabled := v.(bool); enabled {
+			perfCollectionState = "Enabled"
+		}
+	}
```

Comment:
Hi @katbyte , there is no support for SDK or even official documentation from Microsoft yet. Is it ok to still merge this PR ?
I am trying to get hold of Microsoft prod team but as mentioned there is no ETA on it yet

Reply:
Please see the linked issue below for more details, but we typically refrain from merging anything that is not currently supported by the SDK as it can become problematic down the line.

Reply:
Thanks @katbyte ! will wait for SDK and official documentation release